### DISCUSSION TITLE
A couple of smol suggestions (with pictures :D)

### DISCUSSION THREAD

**Discussion by SnailPerson** [bitbof/klecks#45] [Ideas]
![fill alpha concept](https://user-images.githubusercontent.com/97041048/185580117-5ad52d15-46bf-49e5-a972-abf5df54d8e1.png)
Fill-tool with Alpha paint! Selecting "Eraser" enables this feature, opacity would influence how much alpha is added (similar to shape eraser opacity). Tolerance works the same.

![selection tool concept](https://user-images.githubusercontent.com/97041048/185580715-5ad630b7-7dcc-47e8-a5e4-da938d40c1b0.png)
Basic selection tool, explanation and functions in the picture above 

![Bigger stuff](https://user-images.githubusercontent.com/97041048/185591467-7d181ddc-2aec-43dd-b7cb-3cd0cc410662.png)
This always bothered me, especially on smaller screens. precise scaling and perspective and the whole Editing page in general could use a bigger interface. This especially helps when trying to scale layers with small content. 

![Shape lock alpha](https://user-images.githubusercontent.com/97041048/185595085-7edf261d-84a4-4d40-be14-a8606dc77893.png)
Almost every tool has Lock Alpha. But the shape tool does unfortunately not. Having L.A. on the shape tool offers cleaner geometrical drawing. For example one wants to draw a tin can, but getting those curved lines is difficult. Using a circle tool with L.A. can help in this scenario.

Thanks for reading!

**Comment by bitbof** [comment]
Hi, thanks for the suggestions. I love them all, and your take on the logo. Selection and larger/zoomable edit previews are big todos definitely! The other ones should be possible to implement quickly.

**Comment by bitbof** [comment]
added:
- Eraser option for Paint Bucket
- Lock Alpha option for Shape Tool

Selection and better previews are on the agenda, but that will take a while.

**Comment by SnailPerson** [comment]
Pog, thank you so much!

**Comment by bitbof** [comment]
Still missing:

- Larger dialog windows for edit functions. Pretty uncontroversial, should be added.
- A quicker copy paste, without those dialogs. Not sure how to prevent conflict with existing clipboard dialog. Maybe with an active selection it doesn't use the dialog but that could be confusing.
